metadata-ingestion/tests/unit/schema/test_json_schema_util.py

```diff
@@ -374,6 +379,8 @@ def test_nested_arrays():
     expected_field_paths: List[str] = [
         "[version=2.0].[type=NestedArray].[type=array].[type=array].[type=Foo].ar",
         "[version=2.0].[type=NestedArray].[type=array].[type=array].[type=Foo].ar.[type=integer].a",
+        "[version=2.0].[type=NestedArray].[type=array].[type=array].ar",
+        "[version=2.0].[type=NestedArray].[type=array].ar",
```

Comment:
Curious, have you verified on Datahub UI whether descriptions on such complex fields are represented there correctly ?

For example, how does SchemaMetadata aspect generated for below json schema look like when ingested in DataHub ? Is the description for each level captured and represented correctly on UI?
```
{
  "type": "object",
  "title": "NestedArray",
  "namespace": "com.linkedin",
  "properties": {
    "ar": {
      "type": "array",
      "description": "This is description of array ar",
      "items": {
        "type": "array",
        "description": "This is description of inner array of ar",
        "items": {
          "type": "object",
          "title": "Foo",
          "description": "This is description of foo object",
          "properties": {
            "a": {
              "type": "integer",
              "description": "description of integer a"
            }
          }
        }
      }
    }
  }
}
```

Reply:
Hi @mayurinehate ,
Is this expected UI result you are looking for ?
Let me know if there any discrepancies  between UI result i shared and your expected result. 

<img width="1070" alt="abcScreenshot 2024-02-22 at 1 05 07 PM" src="https://github.com/datahub-project/datahub/assets/159135491/df95fdf0-e129-4fbc-8706-affdbde160a5">

Reply:
Hi @mayurinehate 

Is this the expected UI result for the sample nested array you have shared? 
Please let me know if there is any discrepancies between your expected result and UI result i shared.  

<img width="1070" alt="abcScreenshot 2024-02-22 at 1 05 07 PM" src="https://github.com/datahub-project/datahub/assets/156416042/f317418e-ca5f-41c3-bfca-e800d99a1ea5">

Reply:
This does not look right on UI. What was the original behavior before the changes in this PR ? I would image, it only showed the collapsible `ar[][]` and not the last two entries of `ar[][]` and `ar[]`. Is that right ?

Reply:
Hey, My apologies.
yes, collapse is not available for other two entries. I believe you require the nested collapses for all these entries based on the input nested json. Please correct me if i misunderstood. 
Let me fix this up and ping you.